rfcs/configure-file-name-from-block-meta.md

```diff
@@ -0,0 +1,82 @@
+Some extensions may map to multiple languages, such as Perl and Prolog both using `.pl`.
+
+Roughly:
+
+```diff
+- filename: `${index}.${block.lang.trim().split(" ")[0]}`,
++ filename: titleFromMeta(block) ?? `${index}.${block.lang.trim().split(" ")[0]}`,
+```
+
+### Name Uniqueness
+
+Codeblocks must have unique file paths for ESLint processing.
+If a codeblock has the same file path as a previously processed codeblock, processing should throw an error.
+
+If multiple code blocks require the same _file_ name, they can be given different _directory paths_ to ensure uniqueness:
+
+````md
+```json title="example-1/package.json"
+{}
+```
+
+```json title="example-2/package.json"
+{}
+```
+````
+
+### Parsing Meta
+
+There is no unified standard in the ecosystem for parsing codeblock metadata in Markdown.
+The syntax has roughly converged on the syntax looking like <code>\`\`\`lang key="value"</code>, and to a lesser extent, using `title` as the prop name.
+
+-   [Docusaurus's `codeBlockTitleRegex`](https://github.com/facebook/docusaurus/blob/7650829e913ec4bb1263d855719779f6b97066b6/packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts#L12): only supports a [code title prop](https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field) like <code>\`\`\`jsx title="/src/components/HelloCodeTitle.js"</code>
+-   Gatsby plugins such as [`gatsby-remark-prismjs`](https://www.gatsbyjs.com/plugins/gatsby-remark-prismjs) rely on a syntax like <code>\`\`\`javascript{numberLines: true}`</code>.
+    -   Separately, [`gatsby-remark-code-titles`](https://www.gatsbyjs.com/plugins/gatsby-remark-code-titles) allows a syntax like <code>\`\`\`js:title=example-file.js</code>.
+-   [`rehype-mdx-code-props`](https://github.com/remcohaszing/rehype-mdx-code-props): generally handles parsing of the raw `meta` text from the rehype AST.
+    It specifies syntax like <code>\`\`\`js copy filename="awesome.js"</code>, with a suggestion of `filename` for just the file's name.
+    -   [`remark-mdx-code-meta`](https://github.com/remcohaszing/remark-mdx-code-meta) is [referenced in the mdxjs.com `meta` docs](https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field), but was deprecated with a notice to use `rehype-mdx-code-props` instead.
+        It also specified syntax like <code>\`\`\`js copy filename="awesome.js" onUsage={props.beAwesome} {...props}</code>.
+-   [`remark-fenced-props`](https://github.com/shawnbot/remark-fenced-props): A proof-of-concept that augments Remark's codeblock parsing with arbitrary MDX props, written to support [mdx-js/mdx/issues/702](https://github.com/mdx-js/mdx/issues/702).
+    It only specifies syntax like <code>\`\`\`jsx live style={{border: '1px solid red'}}</code>
+
+This RFC chooses `title` over alternatives such as `filename`.
+`title` appears to be the closest to a "popular" choice in existing projects today.
```

Comment:
cc @Josh-Cena from #208 - this looks like something you might have context / opinions on? 😄

Reply:
Mmm, I don't know much outside of Docusaurus/MDN markdown, and the two projects have exactly disjoint sets of meta attributes...

Reply:
Given that, I don't think using `title` is a good idea. `filename` seems like a safer choice. We could also do `eslint-filename` as a completely safe choice, assuming no Markdown parser will balk at unknown attributes.

(We could also use `filename` unless `eslint-filename` is present.)

Reply:
I don't follow. Docusaurus is by far the most popular solution out of these. Is there a reason you're not fond of `title` for safety?

Personally, I actually do think `filename` makes more sense as a technical descriptor. "Title" to me reads as a display title, which isn't always the same as the file name. So I suppose I'm asking more out of thoroughness, not disagreement.

Reply:
Docusaurus is just one use case for Markdown, producing online documentation, but that's not the only use case. I use Markdown to write books and blog posts, and I often feed that Markdown into different tools (including this one).

So, I'm looking at this from a general "what would most people writing Markdown expect?" perspective. This plugin needs to consider the whole universe of Markdown use cases.